neuraxle/base.py

```diff
@@ -2423,6 +2486,34 @@ def get_hyperparams_space(self):
 
         return all_hyperparams.to_flat()
 
+    def update_hyperparams_space(self, hyperparams_space: Union[HyperparameterSpace, OrderedDict, dict]) -> BaseStep:
+        """
+        Update the steps hyperparameters without removing the already-set hyperparameters.
+        Please refer to :func:`~BaseStep.update_hyperparams`.
+
+        :param hyperparams_space: hyperparams_space to update
+        :type hyperparams_space: HyperparameterSamples
+        :return: step
+        :rtype: BaseStep
+
+        .. seealso::
+            :func:`~BaseStep.update_hyperparams`,
+            :class:`HyperparameterSamples`
+        """
+        self.is_invalidated = True
+
+        hyperparams_space: HyperparameterSpace = HyperparameterSpace(hyperparams_space).to_nested_dict()
+
+        remainders = dict()
+        for name, hparams_space in hyperparams_space.items():
+            if name in self.steps.keys():
+                self.steps[name].update_hyperparams_space(HyperparameterSamples(hparams_space))
```

Comment:
Note that the update method looks rlly much like the add or set method. Could use some sort of apply here to combine boths, for getters and for setters, as well for non-space methods (e.g.: update_hyperparams). This would simplify code in the future. not urgent, its just a comment.

Reply:
You can resolve this. This is an issue.